### PR TITLE
EIP 191

### PR DESCRIPTION
**Pull request by seunlanlege**
implemented EIP-191 signing (`personal_sign191`)

### CODE REVIEW

**Comment by 5chdn** [comment]
please rebase on master @seunlanlege 

please review this well

**Comment by seunlanlege** [comment]
actually this depends on #9631

**Comment by tomusdrw** [comment]
Unreviewable in current state, we need to get #9631 merged first.

**Comment by ordian** [comment]
please rebase on master

**Comment by 5chdn** [comment]
#9631 is merged, please rebase

**Comment by 5chdn** [comment]
does not compile

**Comment by seunlanlege** [comment]
rebase went wrong, fixed it now

File: rpc/src/v1/types/eip191.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! EIP-191 specific types
+use serde::{Deserialize, Deserializer};
+use serde::de;
+use v1::types::{H160, Bytes};
+
+pub enum EIP191Version {
```
Comment (ordian) [inline rpc/src/v1/types/eip191.rs]: Please document all public types/fields/functions

File: rpc/src/v1/helpers/eip191.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+use eip712::{hash_structured_data, EIP712};
+use serde_json::{Value, from_value};
+use v1::helpers::errors;
+use jsonrpc_core::Error;
+use v1::helpers::dispatch::eth_data_hash;
+use hash::keccak;
+use std::fmt::Display;
+use ethereum_types::H256;
+
+pub fn decode_message(version: EIP191Version, data: Value) -> Result<H256, Error> {
```
Comment (ordian) [inline rpc/src/v1/helpers/eip191.rs]: I find this name slightly confusing (it also hashes the data), but don't have a better suggestion either

**Comment by Tbaut** [comment]
Can you please add the function to the jsonrpc repo just like with eip712?

File: rpc/src/v1/types/confirmations.rs [hunk 3]
```diff
@@ -94,6 +105,26 @@ impl fmt::Display for SignRequest {
...
+			data: tuple.1,
+		}
+	}
+}
+
+impl fmt::Display for EthSignRequest {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		write!(
+			f,
+			"sign 0x{} with {}",
```
Comment (tomusdrw) [inline rpc/src/v1/types/confirmations.rs]: Should we change the previous implementation to distinguish between eip191 and this?

File: rpc/src/v1/types/eip191.rs (hunk 1 above)
Comment (tomusdrw) [reply rpc/src/v1/types/eip191.rs]: missing docs for variants

**Comment by seunlanlege** [comment]
@tomusdrw 

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-191.md

**Comment by seunlanlege** [comment]
> Is `personal_sign191` the method name other implementations are going to use? I couldn't find anything on it.

In clef, it's currently implemented as `signData`,

https://github.com/ethereum/go-ethereum/pull/17789/files#diff-f8316f6ee7e75303007a55a1cde29245R50

**Pull request merged**